services/management/dbaas/psmdb_cluster_service.go

```diff
@@ -0,0 +1,193 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package dbaas
+
+import (
+	"context"
+	"fmt"
+
+	dbaascontrollerv1beta1 "github.com/percona-platform/dbaas-api/gen/controller"
+	dbaasv1beta1 "github.com/percona/pmm/api/managementpb/dbaas"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/models"
+)
+
+// PSMDBClusterService implements PSMDBClusterServer methods.
+type PSMDBClusterService struct {
+	db               *reform.DB
+	l                *logrus.Entry
+	controllerClient dbaasClient
+}
+
+// NewPSMDBClusterService creates PSMDB Service.
+func NewPSMDBClusterService(db *reform.DB, client dbaasClient) dbaasv1beta1.PSMDBClusterServer {
+	l := logrus.WithField("component", "xtradb_cluster")
+	return &PSMDBClusterService{db: db, l: l, controllerClient: client}
+}
+
+// ListPSMDBClusters returns a list of all PSMDB clusters.
+func (s PSMDBClusterService) ListPSMDBClusters(ctx context.Context, req *dbaasv1beta1.ListPSMDBClustersRequest) (*dbaasv1beta1.ListPSMDBClustersResponse, error) {
+	kubernetesCluster, err := models.FindKubernetesClusterByName(s.db.Querier, req.KubernetesClusterName)
```

Comment:
What happens if the `req.KubernetesClusterName` belongs to a different cluster type? 
For example, if the name belongs to a PXC cluster? Should we check that the type is correct?

Reply:
Kubernetes cluster can't belong to some cluster type. One kubernetes cluster can have a lot of different DB clusters.